src/main/java/org/jabref/gui/keyboard/EmacsKeyBindings.java

```diff
@@ -0,0 +1,103 @@
+package org.jabref.gui.keyboard;
+
+import java.util.Optional;
+
+import javafx.scene.Scene;
+import javafx.scene.control.TextInputControl;
+import javafx.scene.input.KeyEvent;
+
+import org.jabref.Globals;
+import org.jabref.logic.util.strings.EmacsStringManipulator;
+import org.jabref.model.util.ResultingEmacsState;
+import org.jabref.preferences.JabRefPreferences;
+
+public class EmacsKeyBindings {
+
+    public static void executeEmacs(Scene scene, KeyEvent event) {
+        boolean EmacsFlag = Globals.prefs.getBoolean(JabRefPreferences.EDITOR_EMACS_KEYBINDINGS);
```

Comment:
Please start all variables with lower case names. Also applies to CAFlag -- maybe rename it to `flagCA`.